app/code/Magento/Cms/view/adminhtml/web/js/folder-tree.js

```diff
@@ -78,30 +79,92 @@ define([
+                var folderEl = $('[data-id="' + path.pop() + '"]');
 
-                    // if folder doesn't exist, select the last opened folder
-                    if (!folderEl.length) {
-                        tree.jstree('select_node', lastExistentFolderEl);
+                // if folder doesn't exist, select the last opened folder
+                if (!folderEl.length) {
+                    tree.jstree('select_node', lastExistentFolderEl);
 
-                        return;
-                    }
+                    return;
+                }
+
+                lastExistentFolderEl = folderEl;
+
+                if (path.length) {
+                    tree.jstree('open_node', folderEl, recursiveOpen);
+                } else {
+                    tree.jstree('open_node', folderEl, function () {
+                        tree.jstree('select_node', folderEl);
+                    });
+                }
+            },
+
+            /**
+             * Encode path value
+             *
+             * @param {String} val
+             */
+            _encodePathId = function (val) {
+                return Base64.encode(val)
+                    .replace(/\+/g, ':')
+                    .replace(/\//g, '_')
+                    .replace(/=/g, '-');
+            },
+
+            /**
+             * Decode path value
+             *
+             * @param {String} val
+             */
+            _decodePathId = function (val) {
+                return Base64.decode(val)
+                    .replace(/\:/g, '+')
+                    .replace(/\_/g, '/')
+                    .replace(/-/g, '=');
+            },
+
+            /**
+             * Get currentPath decode it returns new path array
+             */
+            _parseCurrentPath = function () {
+                var paths = [],
+                    decodedPath = _decodePathId(window.MediabrowserUtility.pathId.replace(/--|,,/, '==')).split('/');
 
-                    lastExistentFolderEl = folderEl;
+                $.each(decodedPath, function (i, val) {
+                    var isLastElement = i === decodedPath.length - 1;
 
-                    if (path.length) {
-                        tree.jstree('open_node', folderEl, recursiveOpen);
+                    if (isLastElement) {
+                        paths[i] = window.MediabrowserUtility.pathId.replace(',,', '--');
                     } else {
-                        tree.jstree('open_node', folderEl, function () {
-                            tree.jstree('select_node', folderEl);
-                        });
+                        paths[i] = _encodePathId(val);
                     }
-                };
+                });
+                paths.unshift('root');
+                paths.reverse();
+
+                return paths;
+            };
+
+            $(window).on('reload.MediaGallery', function () {
+                pathId =  window.MediabrowserUtility.pathId.replace(',,', '--');
```

Comment:
From our investigation it looks like the `,,` on the pathId (which comes from current_tree_path) just exists when we open the gallery from the uploader button, follow the request example:
`https://magento.local/admin/cms/wysiwyg_images/index/target_element_id/E5U0XX4/store/0/type/image/?isAjax=true&current_tree_path=Y2F0YWxvZy9jYXRlZ29yeQ,,`

But when we open from wysiwyg the problem isn't there:

`https://magento.local/admin/cms/wysiwyg_images/index/current_tree_path/d3lzaXd5Zw--/target_element_id/category_form_description/store/0/type/image/?isAjax=true`

It is probably related to the way the url parameters is being set, can we review the uploader button and fix it to not replace the `-` character? By doing that this replace can be removed.